### DISCUSSION TITLE
Why static filename extension "htm" won't parse but instead of downloaded

### DISCUSSION THREAD

**Discussion by kenjican** [miguelgrinberg/microdot#240] [Q&A] [answered]
Static web service return send_file('path/filename.htm') . Open in Chrome or Firefox, the HTML page  does not be parsed but downloaded.  

     How do I configure microdot to make suffix 'htm' to be parsed ,not downloaded?

**Comment by paroxyste-0** [comment]
Hello, would you like to display the contents of the .htm file as an html page in your browser?

```python
from microdot         import Microdot
from microdot.jinja   import Template

app = Microdot()
Template.initialize(template_dir='templates/')

@app.route('/')
def index(request):
    return Template('filename.htm')
```

Your file must be in ```templates/filename.htm```

**Reply by kenjican** [reply]
Thank you~~ ,I'll try it and come back. It looks like a workable solution.

**Reply by kenjican** [reply]
```
from microdot import Microdot
from jinja import Template

app = Microdot()
Template.initialize(template_dir='/home/kenji/modules/ESP32/config/')

@app.route('/')
def index(request):
    return Template('index.htm')

app.run(port=8888, debug=True)
```
    Code runs with no error in linux , open URL localhost:8888 in Chrome, but Chrome shows blank page and keep loading with no error message. 

    in ESP32 micropython, the error message is ImportError: no module named 'jinja2' . mip.install('Jinja2') doesn't find Jinja2 package.

**Comment by miguelgrinberg** [comment, marked as answer]
The `send_file()` function does not recognize the `htm` extension, so it does not assign the correct content type to it. Try either using `.html`, or else giving the function an explicit content type:

    send_file('file.htm", content_type="text/html")

**Reply by kenjican** [reply]
In ESP32 Micropython v1.23.0 : 
.html works. The tiring thing is all the htm suffix have to revise to html. Add content_type is a better idea.
Thanks~
